.upstream-tests/test/std/containers/views/mdspan/mdspan.ctor/ctad_extents.pass.cpp

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/std/experimental/mdspan>
+#include <cuda/std/cassert>
+
+#define CHECK_MDSPAN(m,d) \
+    static_assert(m.is_exhaustive()); \
+    assert(m.data_handle()  == d.data()); \
+    assert(m.rank()         == 2       ); \
+    assert(m.rank_dynamic() == 2       ); \
+    assert(m.extent(0)      == 64      ); \
+    assert(m.extent(1)      == 128     )
+
+namespace stdex = cuda::std::experimental;
+
+int main(int, char**)
+{
+    // workaround to avoid compilation errors
+    cuda::std::experimental::detail::__make_dynamic_extent<int>();
```

Comment:
@youyu3 Could you please add details of the compiler version and flags here?  This feels very much like a compiler bug.  `__make_dynamic_extent<T>()` literally just returns `T(-1)`.  Its definition should be available, so it shouldn't need to be explicitly instantiated.

Reply:
@youyu3 If `__make_dynamic_extent` doesn't work, we could try this:

```c++
template<class... Integrals>
  requires((is_­convertible_­v<Integrals, size_­t> && ...))
  explicit extents(Integrals...) -> extents<IndexType, std::size_t(indices, -1)...>;
```

What matters here is on the right side of the `->`.  The (C++17) fold expression `size_t(indices, -1)...` expands to something like `size_t(-1)`, `size_t(-1)`, ..., `size_t(-1)`, where the number of `-1`s is `sizeof...(indices)`.

Reply:
Revision:
```c++
template <class... IndexTypes>
extents(IndexTypes...)
 -> extents<std::size_t, std::size_t((IndexTypes(), -1))...>;
```
The comma expression needed an extra (), else it was being interpreted as a cast with two arguments (not a valid cast).

Reply:
Thanks!  Pushed the workaround.